library/src/main/java/com/getstream/sdk/chat/adapter/viewholders/configurators/IndicatorConfigurator.kt

```diff
@@ -0,0 +1,128 @@
+package com.getstream.sdk.chat.adapter.viewholders.configurators
+
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.isGone
+import androidx.core.view.isVisible
+import androidx.core.view.updateLayoutParams
+import com.getstream.sdk.chat.R
+import com.getstream.sdk.chat.adapter.MessageListItem.MessageItem
+import com.getstream.sdk.chat.adapter.MessageViewHolderFactory.Position.BOTTOM
+import com.getstream.sdk.chat.adapter.updateConstraints
+import com.getstream.sdk.chat.adapter.viewholders.getActiveContentViewResId
+import com.getstream.sdk.chat.adapter.viewholders.isDeleted
+import com.getstream.sdk.chat.adapter.viewholders.isEphemeral
+import com.getstream.sdk.chat.adapter.viewholders.isFailed
+import com.getstream.sdk.chat.adapter.viewholders.isInThread
+import com.getstream.sdk.chat.databinding.StreamItemMessageBinding
+import com.getstream.sdk.chat.model.ModelType
+import com.getstream.sdk.chat.utils.LlcMigrationUtils
+import com.getstream.sdk.chat.utils.Utils
+import com.getstream.sdk.chat.view.MessageListView
+import com.getstream.sdk.chat.view.MessageListViewStyle
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.client.models.ChannelUserRead
+
+internal class IndicatorConfigurator(
+    private val binding: StreamItemMessageBinding,
+    private val style: MessageListViewStyle,
+    private val channel: Channel,
+    private val readStateClickListener: MessageListView.ReadStateClickListener
+) : Configurator {
+
+    override fun configure(messageItem: MessageItem) {
+        configDeliveredIndicator(messageItem)
+        configReadIndicator(messageItem)
+        configParamsReadIndicator(messageItem)
+    }
+
+    private fun configDeliveredIndicator(messageItem: MessageItem) {
+        binding.ivDeliver.isVisible = false
+        binding.pbDeliver.isVisible = false
+
+        val message = messageItem.message
+        val lastMessage = LlcMigrationUtils.computeLastMessage(channel)
```

Comment:
I'm not sure if it would be work fine.
I know it was implemented on that way previously, but I am not sure if `channel::messages` is updated (In the future I would like to do it immutable) 
Maybe we could add this info into the `MessageItem` instead of have the channel injected by constructor